module-system/module-implementations/sov-evm/src/query.rs

```diff
@@ -212,13 +212,21 @@ impl<C: sov_modules_api::Context> Evm<C> {
         unimplemented!("eth_sendTransaction not implemented")
     }
 
-    // TODO https://github.com/Sovereign-Labs/sovereign-sdk/issues/502
     #[rpc_method(name = "blockNumber")]
     pub fn block_number(
         &self,
-        _working_set: &mut WorkingSet<C::Storage>,
+        working_set: &mut WorkingSet<C::Storage>,
     ) -> RpcResult<reth_primitives::U256> {
-        unimplemented!("eth_blockNumber not implemented")
+        info!("evm module: eth_blockNumber");
+
+        let block_number = U256::from(
+            self.blocks
+                .last(&mut working_set.accessory_state())
```

Comment:
you can just use `len()`

Reply:
used saturated_sub in case there is no block (low possibility but still)